help/_plugins/SitePostRender.rb

```diff
@@ -21,50 +21,255 @@ def self.process_page(page)
+          clean_text = header_text.downcase.strip
+          header_id = CGI.escape(clean_text.gsub(/\s+/, '-').gsub(/[^\w\-]/, ''))
+
+          # Store the current header's ID in the prefix array
+          prefix[level] = header_id
+
+          # Construct the full hierarchical ID by concatenating IDs for all levels up to the current level
+          full_id = (2..level).map { |l| prefix[l] }.join('--')
+
+          # Assign the generated ID to the header element
+          header['id'] = full_id
+
+          puts "    Found h#{level}: '#{header_text}' -> ID: '#{full_id}'"
         end
 
-        # Determine the level of the header (h2, h3, h4, or h5)
-        level = header.name[1].to_i  # 'h2' -> 2, 'h3' -> 3, etc.
+        # Log the final output being written
+        puts "  Writing updated HTML for page: #{page.path}"
+
+        # Write the updated HTML back to the page
+        page.output = doc.to_html
+      end
+    end
+
+
+    # Generate helpContent.tsx once all pages have been processed
+    Jekyll::Hooks.register :site, :post_render do |site|
+      generate_help_content(site)
+    end
+
+    def self.generate_help_content(site)
+      puts "  Generating helpContent.tsx from rendered HTML pages..."
+
+      output_dir = File.join(site.source, "_src")
+      FileUtils.mkdir_p(output_dir) unless Dir.exist?(output_dir)
+      
+      output_file = File.join(output_dir, "helpContentMap.tsx")
+
+      help_content_tree = generate_help_content_tree()
 
-        # Generate the ID for the current header based on its (cleaned) text
-        clean_text = header_text.downcase.strip
-        header_id = CGI.escape(clean_text.gsub(/\s+/, '-').gsub(/[^\w\-]/, ''))
+      help_content_string = to_ts_object(help_content_tree)
 
-        # Store the current header's ID in the prefix array
-        prefix[level] = header_id
+      imports = [
+        "import type {ReactNode} from 'react';",
+        "import React from 'react';",
+        "import {View} from 'react-native';",
+      ]
 
-        # Construct the full hierarchical ID by concatenating IDs for all levels up to the current level
-        full_id = (2..level).map { |l| prefix[l] }.join('--')
+      # Add conditional imports based on component usage
+      imports << "import BulletList from '@components/SidePanel/HelpComponents/HelpBulletList';" if help_content_string.include?("<BulletList")
+      imports << "import Text from '@components/Text';" if help_content_string.include?("<Text")
+      imports << "import TextLink from '@components/TextLink';" if help_content_string.include?("<TextLink")
+      imports << "import type {ThemeStyles} from '@styles/index';"
 
-        # Assign the generated ID to the header element
-        header['id'] = full_id
+      # Join the imports
+      import_block = imports.join("\n")
 
-        puts "    Found h#{level}: '#{header_text}' -> ID: '#{full_id}'"
+      ts_output = <<~TS
```

Comment:
This TS conversion becomes to difficult to follow, is there a way we can push it as a separate block/split into methods?

Reply:
Done